herbert_core/utilities/strings/str_make_cellstr.m

```diff
@@ -1,54 +1,67 @@
-function [ok,cout]=str_make_cellstr(varargin)
-% Take a set of arguments and try to make a cellstr of strings from the contents
+function [ok, cstr] = str_make_cellstr (varargin)
+% Make a cellstr of character vectors from a set of input arguments
+%
+%   >> [ok, cstr] = str_make_cellstr (c1, c2, c3,...)
+%
+% Differs from str_make_cellstr_trim, which in addition trims leading and
+% trailing whitespace and also removes empty entries.
+%
+% See also str_make_cellstr_trim
 %
-%   >> [ok,cout] = str_make_cellstr(c1,c2,c3,...)
 %
 % Input:
 % ------
-%   c1,c2,c3,...    Two-dimensional character arrays or cell arrays of strings
+%   c1,c2,c3,...    Input text: each can be one of:
+%                   - Character vector (i.e. row vector of characters length >= 0
+%                     or the empty character array, '')
+%                   - Two-dimensional character array
+%                   - Cell array of character vectors
+%                   - strings or string array (Matlab release R2017a onwards)
 %
 % Output:
 % -------
-%   ok              =true if valid input (could all be empty)
-%   cout            Column vector cellstr
-%                  Trailing whitespace is removed from all strings
-%                 (This is for consistency with what cellstr(str) does
+%   ok              Logical scalar:
+%                   - true if all input arguments were convertible to a cell
+%                     array of character vectors
+%                   - false if conversion not possible
+%
+%   cstr            Column vector cell array of character vectors.
+%                   Uses the Matlab function cellstr to perform the conversion
+%                   on each input argument.
+%                   Trailing whitespace is removed from character vectors and 
+%                   character vectors created from 2D character arrays, but not
+%                   from Matlab string objects.
+%
+%                   If ok is false (i.e. conversion of all input arguments to 
+%                   character vectors was not possible), then false.
 
 
-narg=numel(varargin);
+try
+    % Use Matlab cellstr to convert to each input argument to a cell array of
+    % character vectors, if possible.
+    cstr_arr = cellfun(@cellstr, varargin, 'UniformOutput', false);
 
-% Get number of strings
-n=zeros(narg,1);
-for i=1:narg
-    if iscellstr(varargin{i})
-        n(i)=numel(varargin{i});
-    elseif ischar(varargin{i}) && numel(size(varargin{i}))==2
-        if ~isempty(varargin{i})
-            n(i)=size(varargin{i},1);
-        else
-            n(i)=numel(cellstr(varargin{i}));   % so the empty string is stored
-        end
-    else
-        ok=false;
-        cout=cell(0,1);
-        return
+    % There are cases that cellstr returns but which are not a cell array of
+    % character vectors. An example is var = {['Pete';'Bob ']}; this results in
```

Comment:
I avoided using string(var) because then these functions would not work with Matlab prior to R2016b. That is not a problem for PACE, but I have some legacy use of these utilities outside Horace that I don't want to get into having to edit just yet. I'll do that when a better refactor is done of these and other Herbert utilities (ticket #1485) post construction. I'll be in a better position to find the time then to refactor those older bits of code once the end-of-project has passed.